src/renderer/components/editorWithTabs/sourceCode.vue

```diff
@@ -132,39 +133,38 @@
           }, 1000)
         })
       },
-      // A new file was opened or new tab was added.
-      setMarkdown ({ id, markdown }) {
-        this.prepareTabSwitch()
-
-        const { editor } = this
-        editor.setValue(markdown)
-        // NOTE: Don't set the cursor because we load a new file.
-        setCursorAtLastLine(editor)
-        this.tabId = id
-      },
       // Another tab was selected - only listen to get changes but don't set history or other things.
-      handleMarkdownChange ({ id, markdown, cursor, renderCursor, history }) {
+      handleFileChange ({ id, markdown, cursor }) {
         this.prepareTabSwitch()
 
         const { editor } = this
-        editor.setValue(markdown)
+        if (typeof markdown === 'string') {
+          editor.setValue(markdown)
+        }
         // Cursor is null when loading a file or creating a new tab in source code mode.
         if (cursor) {
-          editor.setCursor(cursor)
+          const { anchor, focus } = cursor
+          editor.setSelection(anchor, focus, { scroll: true }) // Scroll the focus into view.
         } else {
           setCursorAtLastLine(editor)
         }
         this.tabId = id
       },
       // Get markdown and cursor from CodeMirror.
       getMarkdownAndCursor (cm) {
-        let cursor = cm.getCursor()
+        let focus = cm.getCursor('head')
+        let anchor = cm.getCursor('anchor')
         const markdown = cm.getValue()
-        const line = cm.getLine(cursor.line)
-        const preLine = cm.getLine(cursor.line - 1)
-        const nextLine = cm.getLine(cursor.line + 1)
-        cursor = adjustCursor(cursor, preLine, line, nextLine)
-        return { cursor, markdown }
+        const adjCursor = cursor => {
```

Comment:
Yeh, as I comment in this PR, we need to adjustCursor the cursor which get from the search results. Because we can not always find a correct position in preview mode according by the search results.

So, Can we get the previous line and next line of search result line?

Reply:
I think this is not a problem and when searching e.g. for a code block we can safely jump into the next line or code block content. I think it's not a big problem and sufficient to correct find raw text matches but the given cursor from ripgrep may be also invalid due files changes.

Did you tested search of an image link/test?

Reply:
>Did you tested search of an image link/test?

No, ripgrep can search in the image link, but preview mode didn't support search in the image link any more, maybe we can just scroll to the image, and select the image, when click the search result.

Reply:
I found another issue, ripgrep search on the disk, but Mark Text maybe not save the markdown content immediately to the disk. So the click search result the cursor maybe not found in the already not saved tab.

Reply:
>  maybe we can just scroll to the image, and select the image, when click the search result.

That's actually a good idea/improvement.

> I found another issue, ripgrep search on the disk, but Mark Text maybe not save the markdown content immediately to the disk. So the click search result the cursor maybe not found in the already not saved tab

Please see below.